src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -41,17 +42,58 @@ public void init() {
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Map<String, String> roomProperties = readJSONFromRequest(request);
 
-    // Save player roomId to datastore.
+    // Save roomId to datastore.
     Entity roomEntity = new Entity("Room");
     roomEntity.setProperty("roomId", roomProperties.get("roomId"));
+    roomEntity.setProperty("userId", roomProperties.get("userId"));
     
     datastore.put(roomEntity);
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Room");
+    PreparedQuery results = datastore.prepare(query);
+    
+    Map<String, String[]> roomIdParamMap = request.getParameterMap();
+
+    for (String roomId : roomIdParamMap.keySet()) {
+    String[] idArray = (String[]) roomIdParamMap.get(roomId);
```

Comment:
this line should be indented (try running the Java formatter, might catch other formatting things I missed(